File: ISAproject/src/main/java/com/example/ISAproject/model/Stuff.java
package com.example.ISAproject.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.List;

import javax.persistence.*;

@Entity
public class Stuff extends User
{
	
	// Vezano je za prvo logovanje
	@Column
	private boolean isFirstLogin;
	
	//Jedan clan osoblja vezan je za jedan centar
	@OneToOne
    private BloodCenter bloodCenter;

	//Dodata je veza sa OSOBLJEM CENTRA-- Jedan clan osoblja ucestvuje u vise termina
	@OneToMany(cascade = CascadeType.ALL,orphanRemoval = true)
	@JsonIgnore
	private List<DonationTerms> terms;



	
	public Stuff(String username, String password, String email,String firstName, String lastName,
			String mobile, String adress, String city, String state, String jmbg, String sex, String profession,
			String organizationInformation, boolean enabled, String role, List<Authority> authorities,
				 boolean firstLogin,BloodCenter bloodCenter)
	{
		super();
		setUsername(username);
		setPassword(password);
		setEmail(email);
		setFirstName(firstName);
		setLastName(lastName);
		setMobile(mobile);
		setAdress(adress);
		setCity(city);
		setState(state);
		setJmbg(jmbg);
		setSex(sex);
		setProfession(profession);
		setOrganizationInformation(organizationInformation);
		setEnabled(enabled);
		setRole(role);
		setAuthorities(authorities);
		this.bloodCenter = bloodCenter;
		this.isFirstLogin = firstLogin;
		
	}
	
	public Stuff()
	{
		
	}

	public Stuff(String username, String password, String email,String firstName, String lastName,
				 String mobile, String adress, String city, String state, String jmbg, String sex, String profession,
				 String organizationInformation, boolean enabled, String role, List<Authority> authorities, boolean firstLogin)
	{
		super();
		setUsername(username);
		setPassword(password);
		setEmail(email);
		setFirstName(firstName);
		setLastName(lastName);
		setMobile(mobile);
		setAdress(adress);
		setCity(city);
		setState(state);
		setJmbg(jmbg);
		setSex(sex);
		setProfession(profession);
		setOrganizationInformation(organizationInformation);
		setEnabled(enabled);
		setRole(role);
		setAuthorities(authorities);
		this.isFirstLogin = firstLogin;


	}

	//Geteri i Seteri
	public BloodCenter getBloodCenter() {
		return bloodCenter;
	}

	public void setBloodCenter(BloodCenter bloodCenter) {
		this.bloodCenter = bloodCenter;
	}

	public boolean isFirstLogin()
	{
		return this.isFirstLogin;
	}
	public void setFirstLogin(boolean firstLogin)
	{
		this.isFirstLogin = firstLogin;
	}
	
	
}

File: ISAfrontend/src/app/model/stuff.ts
import { BloodCenter } from "./bloodCenter";

interface StuffInterface{
    id:number;
    username: string ;
    password: string ;
    email: string;
    firstName: string;
    lastName: string;
    mobile:string;
    state: string;
    city: string;
    adress: string;
    profession: string;
    organizationInformation: string;
    enabled:boolean;
    //fali
    bloodCenter: BloodCenter;
    firstLogin: boolean;
    
}
export class Stuff implements StuffInterface{
    id:number;
    firstName: string;
    lastName: string;
    email: string;
    username: string ;
    password: string ;
    mobile:string;
    state: string;
    city: string;
    adress: string;
    profession: string;
    organizationInformation: string;
    enabled:boolean;
    bloodCenter: BloodCenter;
    firstLogin: boolean;

    constructor(obj:StuffInterface){
        this.id=obj.id;
        this.firstName=obj.firstName;
        this.lastName=obj.lastName;
        this.email=obj.email;
        this.username=obj.username;
        this.password=obj.password;
        this.mobile = obj.mobile;
        this.state = obj.state;
        this.city = obj.city;
        this.adress = obj.adress;
        this.profession = obj.profession;
        this.organizationInformation = obj.organizationInformation;
        this.enabled = obj.enabled;
        this.bloodCenter = obj.bloodCenter;
        this.firstLogin = obj.firstLogin;
  
    } 
}